### ISSUE TITLE
Add deprecation notice to old icon components

### ISSUE THREAD

**Issue by georgewrmarshall** [MetaMask/metamask-extension#17447]
### Description
In an effort to improve new design system component adoption and to prevent any old icons from continuing to be used we could add a JS doc deprecation notice to the icon components in `ui/components/ui/icon`. We may then be able to add an es-lint rule to highlight deprecated components in the code base in another issue.

Deprecation notice to add at the above the component of all icon files:

e.g. in `ui/components/ui/icon/approve-icon.component.js`

```
import React from 'react';
import PropTypes from 'prop-types';

/**
 * @deprecated This has been deprecated in favour of the `<Icon />` component in ./ui/components/component-library/icon/icon.js
 *
 * See storybook documentation for Icon here https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-icon--default-story#icon
 *
 */

const Approve = ({ className, size, color }) => (
  <svg
    className={className}
    width={size}
```

In VS code

![Screenshot 2023-02-09 at 1 30 28 PM](https://user-images.githubusercontent.com/8112138/217943728-1eccae02-8f6f-4a6c-9d66-487159f2f003.png)

<img width="1439" alt="Screenshot 2023-01-26 at 4 16 15 PM" src="https://user-images.githubusercontent.com/8112138/214979078-3b20d360-aeb8-4624-a372-8068b86acbc9.png">

### Technical Details
- Add JS Doc deprecation notice to all icon components in `ui/components/ui/icon`
- Add deprecation notice to `icon.stories.js`

### Acceptance Criteria
- There is a deprecation notice in all icon components in `ui/components/ui/icon`
- There is a deprecation notice in storybook on the old icon story

**Comment by yogesh123gn** [comment]
Hi! I would like to contribute on this issue. I'm actually a beginner, so can you please explain me how to do it ?

**Comment by georgewrmarshall** [comment]
Hey @yogesh123gn, a great start would to be to try and get the repo up and running on your local. Here are the contributor docs https://github.com/MetaMask/metamask-extension#contributing

**Comment by yogesh123gn** [comment]
Thanks alot.

**Comment by chasingtherain** [comment]
Hi @georgewrmarshall 

Are there any other beginner friend tickets you have on hand?

I would like to be assigned a simple issue to work on, thanks!

**Comment by georgewrmarshall** [comment]
Hey @chasingtherain, unfortunately our "good first issues" https://github.com/MetaMask/metamask-extension/labels/good-first-issue is a little lacking but we are working on replacing font-awesome icons(https://github.com/MetaMask/metamask-extension/issues/17448, https://github.com/MetaMask/metamask-extension/issues/17475) with our new [Icon component](https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-icon--default-story#icon) you could pick a font-awesome icon from [the list I posted here](https://github.com/MetaMask/metamask-extension/issues/17475#issuecomment-1418535872) and swap some icons out with the `Icon` component. Sorry it's not super clear, otherwise I will try and break this ticket up into separate issues.

**Comment by chasingtherain** [comment]
Hi @georgewrmarshall 

There are others who have already expressed interest in working on https://github.com/MetaMask/metamask-extension/issues/17448 and https://github.com/MetaMask/metamask-extension/issues/17475. 

Not to worry, when you create the next issue, could you assign it to me? I should have no problem replacing icons.

Thanks in advance!

**Comment by ayushj1910** [comment]
Hey @georgewrmarshall ,
I would like to work on this. Please assign me this task.

**Comment by georgewrmarshall** [comment]
Assigned thanks @ayushj1910

**Comment by ayushj1910** [comment]
Hey @georgewrmarshall , 
I have created a Pull Request "Fix #17447". Please review it.

**Comment by ayushj1910** [comment]
#17679
